src/main/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/DiagnosticReportMapper.java

```diff
@@ -1,12 +1,124 @@
 package uk.nhs.digital.nhsconnect.lab.results.translator.mapper;
 
+import lombok.RequiredArgsConstructor;
 import org.hl7.fhir.dstu3.model.DiagnosticReport;
+import org.hl7.fhir.dstu3.model.Identifier;
+import org.hl7.fhir.dstu3.model.Observation;
+import org.hl7.fhir.dstu3.model.Organization;
+import org.hl7.fhir.dstu3.model.Patient;
+import org.hl7.fhir.dstu3.model.Practitioner;
+import org.hl7.fhir.dstu3.model.Specimen;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.DiagnosticReportDateIssued;
 import uk.nhs.digital.nhsconnect.lab.results.model.edifact.Message;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.Reference;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.ReportStatusCode;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.segmentgroup.ServiceReportDetails;
+import uk.nhs.digital.nhsconnect.lab.results.utils.ResourceFullUrlGenerator;
+import uk.nhs.digital.nhsconnect.lab.results.utils.TimestampService;
+import uk.nhs.digital.nhsconnect.lab.results.utils.UUIDGenerator;
+
+import java.time.LocalDateTime;
+import java.time.ZonedDateTime;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
 
 @Component
+@RequiredArgsConstructor(onConstructor_ = @Autowired)
 public class DiagnosticReportMapper {
-    public DiagnosticReport map(final Message message) {
-        return new DiagnosticReport();
+    private static final Map<ReportStatusCode, DiagnosticReport.DiagnosticReportStatus> STATUS_MAP = Map.of(
```

Comment:
Maybe we can just add the following import at the top `import org.hl7.fhir.dstu3.model.DiagnosticReport.DiagnosticReportStatus;` and instead of doing `DiagnosticReport.DiagnosticReportStatus` we will be able to just do `DiagnosticReportStatus` to make it more readable. What do you think? :)

Reply:
I've done this (in 3b79c0a), but I'm slightly worried that this may lead to some confusion as we have our own class called `DiagnosticReportStatus` which gets the value of the `STS++UN'` segment.